football_now.rb

```diff
@@ -11,50 +12,69 @@ def start
             response = HTTParty.get(BASE_URL)
             games = response.parsed_response['rss']['channel']['item']
 
-            games.each do |game|
-                puts_game(game)
-            end
+            city_prompt
+
+            print_games(games)
 
         rescue SocketError => e
             puts "Check your internet connection. Error message: #{e.message}".light_red.bold
         end
 
         private
-        
-        def puts_game(game)
-            game_splitted = game['title'].split(' - ')
-            puts %Q(#{ datetime_adjust(game['pubDate']) } #{Paint[game_splitted.last, :white, :italic]} - #{ Paint[colorize_benfica(game_splitted.first), :bold] }\n)
-        end
 
-        def datetime_adjust(date)
-            date_splitted = date.split(' ').drop(1)
+        def city_prompt
+            puts "\nPORTUGUESE SPORTV TV CHANNELS"
+            puts "\nEnter the city to convert games date/times into"
+            puts "\nNOTE: type #{Paint['help', :green]} for the list of cities/timezones\n"
 
-            new_time = offset(date_splitted, 'Berlin')
-            date = date_adjust(date_splitted, new_time)
+            user_input
+        end
 
-            "#{date} #{new_time}"
+        def user_input
+            input = gets.chomp
+            input.downcase == 'help' ? cities_tz_list : @city_tz = find_tzinfo(input.capitalize)
         end
 
-        def offset(date_splitted, city_tz)
-            Time.zone = 'London'
-            game_bst_time = Time.zone.parse("#{date_splitted[2]}-#{date_splitted[1]}-#{date_splitted.first} #{date_splitted.last}")
+        def cities_tz_list
+            ActiveSupport::TimeZone.all.sort_by { |o| o.name }.map {|o| puts o.name} 
+            city_prompt
+        end
+
+        def find_tzinfo(input)
+            ActiveSupport::TimeZone.find_tzinfo(input)
 
-            tz = ActiveSupport::TimeZone.find_tzinfo(city_tz)
-            game_cest_time = game_bst_time.in_time_zone(tz)
+        rescue TZInfo::InvalidTimezoneIdentifier => e
+            puts "\n\n#{Paint['We could not find in out timezone list:', :red]} #{e.message}"
+            city_prompt
+        end
+
+        def print_games(games)
+            puts "\nCity timezone of: #{@city_tz.name}: #{@city_tz.now.strftime('%d/%m %H:%M')}\n\n"
 
-            "#{game_cest_time.strftime('%H:%M')}#{Paint[' •', :green, :blink] if is_live?(game_cest_time, tz)}"
+            games.each do |game|
+                puts_game(game)
+            end
         end
 
-        def is_live?(game_cest_time, tz)
-            (game_cest_time..game_cest_time.change(hour: game_cest_time.hour + 2)).cover? Time.now.in_time_zone(tz)
+        def puts_game(game)
+            game_splitted = game['title'].split(' - ')
```

Comment:
This does not feel right in puts_game imo - it has more of parse vibe to me :)

Reply:
Had a method just doing that was adding mkre lines of code than a one-liner, therefore I went fot this